Framework/CurveFitting/test/Functions/FunctionQDependsTest.h

```diff
@@ -0,0 +1,163 @@
+#ifndef MANTID_CURVEFITTING_FUNCTIONQDEPENDSTEST_H
+#define MANTID_CURVEFITTING_FUNCTIONQDEPENDSTEST_H
+
+// Mantid Coding standars <http://www.mantidproject.org/Coding_Standards>
+
+// Mantid Headers from the same project
+#include "MantidCurveFitting/Functions/FunctionQDepends.h"
+// Mantid headers from other projects
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/ParamFunction.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include "MantidKernel/EmptyValues.h"
+#include "MantidDataHandling/LoadNexus.h"
+// 3rd party library headers
+#include <cxxtest/TestSuite.h>
+// Standard library
+// N/A
+
+using Attr = Mantid::API::IFunction::Attribute;
+
+namespace {
+class ImplementsFunctionQDepends
+    : public Mantid::CurveFitting::Functions::FunctionQDepends {
+
+public:
+  std::string name() const override { return "ImplementsFunctionQDepends"; }
+
+  void function1D(double *out, const double *xValues,
+                  const size_t nData) const override {
+    double Q = this->getAttribute("Q").asDouble();
+    for (size_t i = 0; i < nData; i++) {
+      out[i] = Q * xValues[i];
+    }
+  }
+};
+} // end of namespace
+
+class FunctionQDependsTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static FunctionQDependsTest *createSuite() {
+    return new FunctionQDependsTest();
+  }
+  static void destroySuite(FunctionQDependsTest *suite) { delete suite; }
+
+  void testConstruction() {
+    TS_ASSERT_THROWS_NOTHING(ImplementsFunctionQDepends f);
+  }
+
+  void testSetWorkspace() {
+    double startX{0.0}, endX{1.0};
+    ImplementsFunctionQDepends f;
+    // test with an non matrix workspace
+    TS_ASSERT_THROWS_NOTHING(
+        f.setMatrixWorkspace(this->unsuitableWS(), 0, startX, endX));
```

Comment:
There are scenarios when the spectra of the workspace will not contain the necessary information to extract the value of the momentum transfer. In this case, the user will input the Q value either in one of the available graphical interfaces of Mantid, or in a python script statement. That's the reason Q is exposed to the user as an attribute, instead of declared as a private member. For instance, the user may want to load some cheap & dirty synthetic data from a file to a workspace, but the file does not contain the Q-values. The user then has the option to manually insert these values.